portal/client-services/org.wso2.is.portal.user.client.api/src/main/java/org/wso2/is/portal/user/client/api/IdentityStoreClientServiceImpl.java

```diff
@@ -501,4 +546,200 @@ private RealmService getRealmService() {
+                .collect(Collectors.toList());
+
+        try {
+            getRealmService().getIdentityStore().updateGroupClaims(uniqueGroupId, updatedClaims);
+        } catch (IdentityStoreException | GroupNotFoundException e) {
+            String error = "Failed to updated group profile.";
+            LOGGER.error(error, e);
+            throw new UserPortalUIException(error);
+        }
+    }
+
+    @Override
+    public List<UserListBean> getUserList(int offset, int length, String domainName,
+            List<ClaimConfigEntry> requestedClaims) throws UserPortalUIException {
+        List<User> users;
+        if (length < 0) {
+            length = MAX_RECORD_LENGTH;
+        }
+        try {
+            users = getRealmService().getIdentityStore().listUsers(offset, length, domainName);
+        } catch (IdentityStoreException e) {
+            String error = "Error while retrieving users";
+            LOGGER.error(error, e);
+            throw new UserPortalUIException(error);
+        }
+
+        return generateUserListBean(users, requestedClaims);
+    }
+
+    @Override
+    public List<GroupListUserBean> getUserListForAssignment(int offset, int length, String domainName,
+            List<ClaimConfigEntry> requestedClaims) throws UserPortalUIException {
+        List<User> users;
+        if (length < 0) {
+            length = MAX_RECORD_LENGTH;
+        }
+        try {
+            users = getRealmService().getIdentityStore().listUsers(offset, length, domainName);
+        } catch (IdentityStoreException e) {
+            String error = "Error while retrieving users";
+            LOGGER.error(error, e);
+            throw new UserPortalUIException(error);
+        }
+
+        return generateUserListBeanForGroup(users, requestedClaims);
+    }
+
+    private List<GroupListUserBean> generateUserListBeanForGroup(List<User> users,
+            List<ClaimConfigEntry> requestedClaims) throws UserPortalUIException {
+        List<GroupListUserBean> userList = new ArrayList<>();
+
+        List<MetaClaim> metaClaims = requestedClaims.stream()
+                .map(claimConfigEntry -> new MetaClaim("", claimConfigEntry.getClaimURI()))
+                .collect(Collectors.toList());
+
+        for (User userEntry : users) {
+            List<Claim> userClaims;
+            try {
+                userClaims = userEntry.getClaims(metaClaims);
+
+            } catch (IdentityStoreException | UserNotFoundException e) {
+                String error = "Error while retrieving user data for user :  " + userEntry.getUniqueUserId();
+                LOGGER.error(error, e);
+                throw new UserPortalUIException(error);
+            }
+
+            GroupListUserBean listEntry = new GroupListUserBean();
+            listEntry.setPicture(null);           //set profile picture to be shown.
+            listEntry.setUid(userEntry.getUniqueUserId());
+            listEntry.setUsername(userClaims.get(0).getValue());
```

Comment:
Addressed with iterating over the list and filtering the user claim. Zero was used as it was know there is only one claim and that is user claim as per the profile configuration. As this is not elegant corrected as pointed.